src/estimators/meta_estimators.py

```diff
@@ -0,0 +1,171 @@
+          'MetaEstimator must be used with an AnyDistributionBloomFilter.')
+
+    first_sketch = sketch_list[0]
+
+    for cur_sketch in sketch_list[1:]:
+      first_sketch.assert_compatible(cur_sketch)
+
+  def __call__(self, sketch_list):
+    """Transform AnyDistributionBloomFilters into meta sketch and estimate union."""
+    MetaEstimatorBase._check_compatibility(sketch_list)
+    if not sketch_list:
+      return 0
+
+    meta_sketches = self._transform_adbf_into_meta_sketches(sketch_list)
+    num_ones = self._estimate_num_ones(meta_sketches)
+    adbf_sketch = MetaEstimatorBase._construct_fake_adbf(
+        num_ones, sketch_list[0])
+    return self.adbf_estimator([adbf_sketch])
+
+  def _transform_adbf_into_meta_sketches(self, sketch_list):
+    """Transform AnyDistributionBloomFilters into meta sketch.
+
+    Args:
+      sketch_list: a list of AnyDistributionBloomFilter sketches.
+
+    Returns:
+      A list of meta sketches.
+    """
+    if not sketch_list:
+      return []
+
+    random_seed = sketch_list[0].random_seed
+
+    meta_sketches = []
+    for sketch in sketch_list:
+      meta_sketch = self.meta_sketch_factory(random_seed)
+
+      # Get nonzero indices in sketch and add those indices to the meta_sketch.
+      nonzero_indices = np.flatnonzero(sketch.sketch)
+      meta_sketch.add_ids(nonzero_indices)
+
+      meta_sketches.append(meta_sketch)
+
+    return meta_sketches
+
+  def _estimate_num_ones(self, meta_sketches):
+    """Estimate the number of one registers from the meta sketches.
+
+    Args:
+      meta_sketches: a list of meta sketches.
+
+    Returns:
+      The estimated number of one registers.
+    """
+    # Estimate the number of 1's from the meta sketches.
+    return int(self.meta_sketch_estimator(meta_sketches)[0])
+
+  @classmethod
+  def _construct_fake_adbf(cls, num_ones, template_adbf):
+    """Create a fake AnyDistributionBloomFilter.
+
+    We create a fake ADBF which can be used to estimate the cardinality.
+    The first num_ones of register will be set to 1, and 0 otherwise.
+    As such, we can reuse the FirstMomentEstimator.
+    Note that it is only compatible with estimators that doesn't rely on the
+    per bucket probability.
+    want to be able to reuse the FirstMomentEstimator code here
+    which requires that we input a sketch. In order to handle this,
+    we will fill a new sketch with union_size number of ones in the
+    first n indices. This may break if we use estimators which care
+    about the specific bucket indices. This is not currently supported
+    behavior.
+
+    Args:
+      num_ones: the number of one registers.
+      template_adbf: a template AnyDistributionBloomFilter. The created sketch
+        will be used to estimate the union cardinality.
+
+    Returns:
+      An AnyDistributionBloomFilter, which has the same setting as the
+      template_adbf sketch.
+    """
+    new_sketch = copy.deepcopy(template_adbf)
+    new_sketch.sketch[:num_ones] = 1
```

Comment:
You might want to ensure you are not out of bounds on the array.

Also I'm a bit skeptical of access the sketch data directly, but maybe here it's okay.

Reply:
Done.